### PR TITLE
Unit test for new translation contributor

### LINKED ISSUE #210: Add unit test for first-time contributors
Add unit test for the improvement in #189

### PR DESCRIPTION
**Pull request by trymebytes**
Fixes #210

### CODE REVIEW

File: tests/attendee/attendee-repository.php [hunk 1]
```diff
@@ -206,6 +221,46 @@ public function test_get_events_for_user() {
...
+		$attendee12 = new Attendee( $event1_id, $user2_id );
+		$attendee13 = new Attendee( $event1_id, $user3_id );
+		$this->repository->insert_attendee( $attendee11 );
+		$this->repository->insert_attendee( $attendee12 );
+		$this->repository->insert_attendee( $attendee13 );
+		$event = $this->event_repository->get_event( $event1_id );
+
+		$contributor_ids     = array( $attendee11->user_id(), $attendee12->user_id(), $attendee13->user_id() );
+		$new_contributor_ids = array();
+		$translations_counts = $this->translation_repository->count_translations_before( $contributor_ids, $event->start() );
```
Comment (akirk) [inline tests/attendee/attendee-repository.php]: I know that this is how new contributors (after the performance improvement in #225) is calculated but we should test this inherently.

Thus, can we add a `is_new_contributor()` implementation with this test that uses `count_translations_before` internally?

**Comment by akirk** [comment]
A sample implementation could be like this:
```diff
diff --git a/includes/routes/event/details.php b/includes/routes/event/details.php
--- includes/routes/event/details.php
+++ includes/routes/event/details.php
@@ -73,9 +73,22 @@
 				return ! $attendee->is_contributor();
 			}
 		);
 
+		$new_contributor_ids = array_filter(
+			$contributor_ids,
+			function( $contributor_id ) use ( $event ) {
+				$contributor = new Attendee( $contributor_id );
+				return $contributor->is_new_contributor();
+		} );
+
+		class Attendee {
+			function is_new_contributor() {
+				$event = $this->translation_repository->get_event( $this->event_id );
+				return $this->translation_repository->count_translations_before( [ $this->id ], $event->start() )[$this->id] <= 10;
+			}
+		}
+
		$contributor_ids = array_map(
 			function ( Attendee $contributor ) {
 				return $contributor->user_id();
 			},
 			$contributors
diff --git a/includes/upgrade.php b/includes/upgrade.php
--- includes/upgrade.php
+++ includes/upgrade.php
@@ -46,8 +46,9 @@
 				`translate_event_attendees_id` int(11) unsigned NOT NULL AUTO_INCREMENT,
 				`event_id` int(10) NOT NULL COMMENT 'Post_ID of the translation_event post in the wp_posts table',
 				`user_id` int(10) NOT NULL COMMENT 'ID of the user who is attending the event',
 				`is_host` tinyint(1) default 0 not null comment 'Whether the user is a host of the event',
+				`is_new_contributor` tinyint(1) default 0 not null comment 'Whether the user is a host of the event',
 			PRIMARY KEY (`translate_event_attendees_id`),
 			UNIQUE KEY `event_per_user` (`event_id`,`user_id`),
 			INDEX `user` (`user_id`)
 			) COMMENT='Attendees of events';
```

File: tests/attendee/attendee-repository.php [hunk 2]
```diff
@@ -206,6 +221,46 @@ public function test_get_events_for_user() {
...
+		foreach ( $translations_counts as $user_id => $count ) {
+			if ( $count <= 10 ) {
+				$new_contributor_ids[ $user_id ] = true;
+			}
+		}
+
+		$this->assertCount( 2, $new_contributor_ids );
+		$this->assertTrue( $new_contributor_ids[ $user1_id ] );
+		$this->assertTrue( $new_contributor_ids[ $user2_id ] );
```
Comment (akirk) [inline tests/attendee/attendee-repository.php]: The asserts would then look like this:
```suggestion
		$this->assertTrue( $attendee11->is_new_contributor() );
		$this->assertTrue( $attendee12->is_new_contributor() );
		$this->assertFalse( $attendee13->is_new_contributor() );
```

File: tests/attendee/attendee-repository.php [hunk 3]
```diff
@@ -206,6 +214,36 @@ public function test_get_events_for_user() {
...
+		$attendee12 = new Attendee( $event1_id, $user2_id );
+		$attendee13 = new Attendee( $event1_id, $user3_id );
+		$this->repository->insert_attendee( $attendee11 );
+		$this->repository->insert_attendee( $attendee12 );
+		$this->repository->insert_attendee( $attendee13 );
+
+		$this->assertTrue( $attendee11->is_new_contributor() );
+		$this->assertTrue( $attendee12->is_new_contributor() );
+		$this->assertFalse( $attendee13->is_new_contributor() );
+	}
```
Comment (akirk) [inline tests/attendee/attendee-repository.php:256]: I think it'd be also interesting to test whether a user would be a new contributor for an earlier event.
```suggestion

		$event2_id  = $this->event_factory->create_active( array(), $now->modify( '-1 day' ) );
		$attendee21 = new Attendee( $event2_id, $user1_id );
		$attendee22 = new Attendee( $event2_id, $user2_id );
		$attendee23 = new Attendee( $event2_id, $user3_id );
		$this->repository->insert_attendee( $attendee21 );
		$this->repository->insert_attendee( $attendee22 );
		$this->repository->insert_attendee( $attendee23 );
		$this->assertTrue( $attendee21->is_new_contributor() );
		$this->assertTrue( $attendee22->is_new_contributor() );
		$this->assertTrue( $attendee23->is_new_contributor() );
	}
```

**Review by akirk — APPROVED** [review]
Great! I just had a little suggestion for some more assertions, but already like this approvable.

File: tests/attendee/attendee-repository.php (hunk 3 above)
Comment (trymebytes) [reply tests/attendee/attendee-repository.php:256]: Oh, that's true too, thanks!

**Review by akirk — APPROVED** [review]

**Pull request merged**